index.js

```diff
@@ -679,7 +684,9 @@ class KeyringController extends EventEmitter {
    */
   async restoreKeyring(serialized) {
     const keyring = await this._restoreKeyring(serialized);
-    await this._updateMemStoreKeyrings();
+    if (keyring) {
```

Comment:
Should we add a test for `restoreKeyring` which tests this behavior?

Reply:
I have tested this [here](https://github.com/MetaMask/KeyringController/pull/169/files#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440bR360) by calling `unlockKeyrings` method.

I could not find use of `restoreKeyring` method anywhere, is it added for testing purpose ?

Reply:
Good question. This method dates back long enough that I don't know the reason. I can't think of a use for it.

Reply:
I added test case for `restoreKeyring`